fe4u-boilerplate-main/src/js/Popup/popupTeacherCard.js

```diff
@@ -0,0 +1,100 @@
+/* eslint-disable */
+const L = require('leaflet');
+const {content, popupModalInfo} = require('../Lab2/PopupFunctional')
+
+class popupTeacherCard {
+    constructor() {
+        this.teacher = null;
+        this.favCallBack = null;
+        this.isCliked = false;
+        this.favbtn = document.getElementById('ifavorite');
+        this.closebtn = document.getElementById('close-popup')
+        this.mapbtn = document.getElementById('map-btn')
+
+        this.infoteacherPopup = document.getElementById('infoPopup');
+        this.headerPopup = document.getElementById('headerPopup');
+        this.mapPopup = document.getElementById('map-content');
+        this.favbtn.onclick = () => {
+            if (!this.teacher) return;
+            this.teacher.favorite = !this.teacher.favorite;
+            this.teacher.favorite ? this.favbtn.classList.add('selected') : this.favbtn.classList.remove('selected');
+            if (this.favCallBack) this.favCallBack(this.teacher.favorite);
```

Comment:
All `if` should have curly brackets, even if it contains one operation only, because
1. the syntax without `{}` is hard to read
2. you will change more lines when adding new operations. So you will make later redundant changes in PR